### PR TITLE
ci(postman): Run Postman collections against PRs 

### PR DESCRIPTION
**Pull request by pixincreate**
## Type of Change

- [x] Refactoring
- [x] CI/CD

## Description

This PR aims at running the postman collections against the PRs raised instead of running on schedule on Integ so that any changes that break the main will be caught at PR level.  

Keeping costs in mind, I would suggest running the collections at the time of merge queue rather than running at every commit. 

## Motivation and Context

It becomes a headache to track down the PR that broke the main. So, running tests globally for every PR would fix the issue. 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: .github/workflows/postman-collection-runner.yml [hunk 1]
```diff
@@ -1,68 +1,124 @@
-name: Postman Collection Runner
+name: Run collections against your PRs
 
 on:
   workflow_dispatch:
-  schedule:
-    - cron: "30 0,12 * * *" # Run workflow at 6 AM and 6 PM IST
+  merge_group:
```
Comment (ashokkjag) [inline .github/workflows/postman-collection-runner.yml:5]:
```suggestion
  merge_group:
      types: [checks_requested]
```

File: .github/workflows/postman-collection-runner.yml [hunk 2]
```diff
@@ -1,68 +1,124 @@
...
-          - bluesnap
-          # - braintree
-          - checkout
-          # - iatapay
-          # - multisafepay
-          - nmi
-          - stripe
-          # - trustpay
-          - worldline
+          - aci, checkout, stripe
```
Comment (ashokkjag) [inline .github/workflows/postman-collection-runner.yml:27]: verify that this is what you want? syntax may be wrong here.

File: .github/workflows/postman-collection-runner.yml [hunk 3]
```diff
@@ -1,68 +1,124 @@
...
+        id: config_path
+        shell: bash
+        run: |
+          echo "CONNECTOR_AUTH_FILE_PATH=$HOME/target/test/connector_auth.toml" >> $GITHUB_ENV
+
+      - name: Fetch keys
+        env:
+          TOML_PATH: "./config/development.toml"
+        run: |
+          KEY_LINE=$(grep -E '^\[secrets\]$|^\s*admin_api_key\s*=' "$TOML_PATH")
```
Comment (ashokkjag) [inline .github/workflows/postman-collection-runner.yml:74]: I see `yq` is already available in the ubuntu runner.

you can use 
```bash
> yq '.secrets.admin_api_key' config/development.toml
test_admin
```

File: crates/test_utils/src/main.rs [hunk 4]
```diff
@@ -97,11 +96,11 @@ fn main() {
             }
             // Handle other ConnectorAuthType variants
             _ => {
-                eprintln!("Invalid authentication type.");
+                eprintln!("\x1b[31mInvalid authentication type.\x1b[0m");
```
Comment (ashokkjag) [inline crates/test_utils/src/main.rs:99]: Any reason for adding the ansi codes. these are OK for terminal but not useful for github actions.

**Review by ashokkjag — CHANGES_REQUESTED** [review]

File: crates/test_utils/src/main.rs (hunk 4 above)
Comment (pixincreate) [reply crates/test_utils/src/main.rs:99]: This did work in GitHub actions when I tested it yesterday. I'll remove this for `--color` command in Newman

File: .github/workflows/postman-collection-runner.yml (hunk 2 above)
Comment (ashokkjag) [reply .github/workflows/postman-collection-runner.yml:28]: After discussion, intention is to pass multiple `,` separated values to each job to reduce the startup/runtime.

**Review by ashokkjag — APPROVED** [review]

File: .github/workflows/postman-collection-runner.yml (hunk 2 above)
Comment (pixincreate) [reply .github/workflows/postman-collection-runner.yml:30]: Yea, this reduces startup time and hence allows us to reuse already set-up server for upcoming connectors.   
With this, we could effectively reduce the run time from a mere 16+ minutes to ~7 minutes

**Review by ashokkjag — DISMISSED** [review]

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Review by SanchithHegde — APPROVED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by ashokkjag — APPROVED** [review]

**Pull request merged**